tests/ManiaTemplates.Tests/IntegrationTests/ManialinkEngineTest.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace ManiaTemplates.Tests.IntegrationTests;
+
+public class ManialinkEngineTest
+{
+    private readonly ManiaTemplateEngine _maniaTemplateEngine = new();
+    
+    [Theory]
+    [ClassData(typeof(TestDataProvider))]
+    public void Should_Convert_Templates_To_Result(string template, dynamic data, string expected)

```

Comment:
Does this test actually run? It throws the following on my machine:

```
System.Reflection.TargetException
Object does not match target type.
   at System.Reflection.MethodBase.ValidateInvokeTarget(Object target)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at ManiaTemplates.Lib.ManiaLink.RenderInternalAsync(Object runnable) in ManiaTemplates/src/ManiaTemplates/Lib/ManiaLink.cs:line 92
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at ManiaTemplates.Lib.ManiaLink.Render(Object data) in ManiaTemplates/src/ManiaTemplates/Lib/ManiaLink.cs:line 40
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at ManiaTemplates.ManiaTemplateEngine.Render(String key, Object data, IEnumerable`1 assemblies) in ManiaTemplates/src/ManiaTemplates/ManiaTemplateEngine.cs:line 135
   at System.Dynamic.UpdateDelegates.UpdateAndExecute4[T0,T1,T2,T3,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
   at ManiaTemplates.Tests.IntegrationTests.ManialinkEngineTest.Should_Convert_Templates_To_Result(String template, Object data, String expected) in ManiaTemplates/tests/ManiaTemplates.Tests/IntegrationTests/ManialinkEngineTest.cs:line 14
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
```

Reply:
I can't figure out why this should not work though.

Reply:
Been staring at this for an hour now, doesn't make sense at all. Is it possible it could be a bug in System.Reflection?

Reply:
Ok I figured out the problem. Because when you call the non-async `Render` method, which it does in the test, the underlying method calls RenderInternalAsync with the data as the argument, so the "runnable" variable is now the data. But it should be the instance of the text transformer. Hence the object which TransformText is called upon does not match the type which TransformText was obtained from.

Reply:
It's just missing the creation of a TextTransformer instance. But the sync Render method is literally just a helper method and I'm not sure if its actually necessary.